CUDA : OpenCV requires enabled 'cudev' module from 'opencv_contrib'

i am building opencv c++ on windows using the script here : https://docs.opencv.org/3.4.3/d3/d52/.... I enabled cuda, tests, perf_tests, docs and samples as i wanted to build eveything. but i get this compilation cmake error. I have installed CUDA toolkit 10.1 on my machine. CMake Error at modules/core/CMakeLists.txt:39 (message): CUDA: OpenCV requires enabled 'cudev' module from 'opencv_contrib' repository: https://github.com/opencv/opencv_contrib Call Stack (most recent call first): modules/world/CMakeLists.txt:13 (include) modules/world/CMakeLists.txt:32 (include_one_module)

I believe the script sets opencv_contrib path which has cudev module. Unable to figure out what's going wrong here.
Prior i was able to build it and This started happening since 2 days.

--- a/modules/core/CMakeLists.txt
+++ b/modules/core/CMakeLists.txt
@@ -35,7 +35,7 @@ if(DEFINED WINRT AND NOT DEFINED ENABLE_WINRT_MODE_NATIVE)
 endif()
 
 if(HAVE_CUDA)
-  if(NOT TARGET opencv_cudev)
+  if(NOT HAVE_opencv_cudev)
     message(FATAL_ERROR "CUDA: OpenCV requires enabled 'cudev' module from 'opencv_contrib' repository: https://github.com/opencv/opencv_contrib")
   endif()
   ocv_warnings_disable(CMAKE_CXX_FLAGS -Wundef -Wenum-compare -Wunused-function -Wshadow)

